cloudsim-plus-testbeds/src/main/java/org/cloudsimplus/migration/ExampleMigrationVmWhenCpuMetricIsViolated.java

```diff
@@ -0,0 +1,340 @@
+        for (int i = 0; i < NUMBER_OF_CLOUDLETS_TO_CREATE_BY_VM; i++) {
+            cloudletId = hostingVm.getId() + i;
+            UtilizationModelArithmeticProgression cpuUtilizationModel;
+            if (progressiveCpuUsage) {
+                cpuUtilizationModel
+                        = new UtilizationModelArithmeticProgression(
+                                CLOUDLET_CPU_USAGE_INCREMENT_PER_SECOND,
+                                cloudletInitialCpuUtilizationPercentage);
+            } else {
+                cpuUtilizationModel
+                        = new UtilizationModelArithmeticProgression(0,
+                                cloudletInitialCpuUtilizationPercentage);
+            }
+            cpuUtilizationModel.setMaxResourceUsagePercentage(maxCloudletCpuUtilizationPercentage);
+
+            Cloudlet c
+                    = new CloudletSimple(
+                            cloudletId, CLOUDLET_LENGHT, VM_PES_NUM)
+                    .setCloudletFileSize(CLOUDLET_FILESIZE)
+                    .setCloudletOutputSize(CLOUDLET_OUTPUTSIZE)
+                    .setUtilizationModelCpu(cpuUtilizationModel)
+                    .setUtilizationModelRam(utilizationModelFull)
+                    .setUtilizationModelBw(utilizationModelFull);
+            c.setBroker(broker);
+            list.add(c);
+        }
+
+        broker.submitCloudletList(list);
+        for (Cloudlet c : list) {
+            broker.bindCloudletToVm(c, hostingVm);
+        }
+
+        return list;
+    }
+
+    public static List<Cloudlet> createAndSubmitCloudletsWithDynamicCpuUtilization(
+            double initialCloudletCpuUtilizationPercentage,
+            double maxCloudletCpuUtilizationPercentage,
+            Vm hostingVm,
+            DatacenterBroker broker) {
+        return createAndSubmitCloudlets(
+                initialCloudletCpuUtilizationPercentage,
+                maxCloudletCpuUtilizationPercentage, hostingVm, broker, true);
+    }
+
+    public static List<Cloudlet> createAndSubmitCloudletsWithStaticCpuUtilization(
+            double initialCloudletCpuUtilizationPercentage,
+            Vm hostingVm,
+            DatacenterBroker broker) {
+        return createAndSubmitCloudlets(
+                initialCloudletCpuUtilizationPercentage,
+                initialCloudletCpuUtilizationPercentage,
+                hostingVm, broker, false);
+    }
+
+    private static Datacenter createDatacenter() {
+        ArrayList<PowerHost> hostList = new ArrayList<>();
+        for (int i = 0; i < NUMBER_OF_HOSTS_TO_CREATE; i++) {
+            hostList.add(createHost(i, HOST_NUMBER_OF_PES, HOST_MIPS_BY_PE));
+            Log.printConcatLine("#Created host ", i, " with ", HOST_MIPS_BY_PE, " mips x ", HOST_NUMBER_OF_PES);
+        }
+        Log.printLine();
+
+        DatacenterCharacteristics characteristics
+                = new DatacenterCharacteristicsSimple(hostList)
+                .setCostPerSecond(DATACENTER_COST_PER_CPU)
```

Comment:
If you are not using the costs, remove these setters. Since the cost is irrelevant for the example, setting them just adds complexity to the code, once another developer may ask way cost are being set.